pkg/networkservice/common/clienturl/server_test.go

```diff
@@ -22,84 +22,108 @@ import (
+
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
-	"github.com/stretchr/testify/assert"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
-type testNetworkServiceServer struct {
-	t    *testing.T
-	want *url.URL
+type contextKeyType string
+
+const testDataKey contextKeyType = "testData"
+
+type TestData struct {
+	clientURL *url.URL
 }
 
-func (c *testNetworkServiceServer) Request(ctx context.Context, in *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
-	assert.Equal(c.t, c.want, clienturl.ClientURL(ctx))
-	return in.GetConnection(), nil
+func withTestData(parent context.Context, testData *TestData) context.Context {
+	if parent == nil {
+		parent = context.TODO()
+	}
+	return context.WithValue(parent, testDataKey, testData)
 }
 
-func (c *testNetworkServiceServer) Close(ctx context.Context, _ *networkservice.Connection) (*empty.Empty, error) {
-	assert.Equal(c.t, c.want, clienturl.ClientURL(ctx))
-	return &empty.Empty{}, nil
+func testData(ctx context.Context) *TestData {
+	if rv, ok := ctx.Value(testDataKey).(*TestData); ok {
+		return rv
+	}
+	return nil
 }
 
-var testData = []struct {
-	name  string
-	ctx   context.Context
-	given *url.URL
-	want  *url.URL
-}{
-	{
-		"add client url in empty context",
-		context.Background(),
-		&url.URL{
-			Scheme: "ipv4",
-			Path:   "192.168.0.1",
-		},
-		&url.URL{
-			Scheme: "ipv4",
-			Path:   "192.168.0.1",
-		},
-	},
-	{
-		"overwrite client url",
-		clienturl.WithClientURL(context.Background(), &url.URL{
-			Scheme: "unix",
-			Path:   "/var/run/nse-1.sock",
-		}),
-		&url.URL{
-			Scheme: "ipv4",
-			Path:   "192.168.0.1",
-		},
-		&url.URL{
-			Scheme: "ipv4",
-			Path:   "192.168.0.1",
-		},
-	},
-	{
-		"overwrite client url by nil",
-		clienturl.WithClientURL(context.Background(), &url.URL{
-			Scheme: "unix",
-			Path:   "/var/run/nse-1.sock",
-		}),
-		nil,
-		nil,
-	},
+type testNetworkServiceServer struct{}
+
+func (c *testNetworkServiceServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	testData(ctx).clientURL = clienturl.ClientURL(ctx)
+	return next.Server(ctx).Request(ctx, request)
 }
 
-func Test_clientUrlServer(t *testing.T) {
-	for _, data := range testData {
-		test := data
-		t.Run(test.name, func(t *testing.T) {
-			testServer(test.ctx, test.given, test.want, t)
-		})
+func (c *testNetworkServiceServer) Close(ctx context.Context, conn *networkservice.Connection) (*empty.Empty, error) {
+	testData(ctx).clientURL = clienturl.ClientURL(ctx)
+	return next.Server(ctx).Close(ctx, conn)
+}
+
+func TestAddURLInEmptyContext(t *testing.T) {
+	clientURL := &url.URL{
+		Scheme: "ipv4",
+		Path:   "192.168.0.1",
 	}
+	client := next.NewNetworkServiceServer(clienturl.NewServer(clientURL), &testNetworkServiceServer{})
```

Comment:
Why not use https://github.com/networkservicemesh/sdk/tree/master/pkg/networkservice/utils/checks/checkcontext here?

Reply:
Switched to this utility, thanks